app/policies/admin_profile_policy.rb

```diff
@@ -1,13 +1,31 @@
 # frozen_string_literal: true
 
 class AdminProfilePolicy < ApplicationPolicy
+  def show?
+    admin_only
+  end
+
+  def index?
+    admin_only
+  end
+
+  def destroy?
+    admin_only
+  end
+
   def create?
     admin_only
   end
 
+  def permitted_attributes
+    if user.admin?
+      :id
+    end
+  end
+
   class Scope < Scope
     def resolve
-      return scope.all if user.admin?
+      return scope.kept if user.admin?
```

Comment:
Do we need this in addition to the default scope?

Reply:
Done